src/business/sorting.js

```diff
@@ -0,0 +1,41 @@
+function returnMoreUsed(mealA, mealB) {
+  const mealAUses = mealA.usedCount || 0;
+  const mealBUses = mealB.usedCount || 0;
+
+  return mealBUses - mealAUses;
+}
+
+function sortMostUsed(meals) {
+  return Array.from(meals).sort(returnMoreUsed);
+}
+
+function compareLastUsed(mealA, mealB) {
+  const mealADate = mealA.lastUsed || 0;
+  const mealBDate = mealB.lastUsed || 0;
+
+  return mealBDate - mealADate;
+}
+
+function sortRecentlyPrepared(meals = []) {
+  return Array.from(meals).sort(compareLastUsed);
+}
+
+function compareId(mealA, mealB) {
+  const mealAId = mealA.id || 0;
+  const mealBId = mealB.id || 0;
+
+  return mealAId - mealBId;
+}
+
+function sortById(meals = []) {
+  return Array.from(meals).sort(compareId)
+}
```

Comment:
Very true. if we were going by baby steps, we'd start with what you're suggesting but curry those functions. we could then compose more targeted functions with expressive names by doing things like
```
const sortByName = sortBy("name");
const sortListByName = map(sortByName);
```

I just skipped that first step :) I could just has easily have skipped the second step instead, or in addition, and just composed sortListByName all in one place. It's kind of all just algebra at that point.

Reply:
Yeah - just a matter of what you think works best. I, personally, am always thinking about the eventual filesize of my built package - and while I don't always optimize for it, if I have the option of defining 1 function instead of 3 without sacrificing readability, I typically go for it just to save bytes.

It matters less for an application like ours - while it would matter more for something rendering on mobile, where every kilobyte matters in load time.

Reply:
I actually agree with you. My default is to define fewer functions, if readability is all the same. But i'm finding that doing things like this is a very useful exercise to get myself better at composing functions. The process has not been painless, so i need all the practice I can get.